mlton/backend/limit-check.fun

```diff
@@ -844,17 +848,21 @@ fun transform (Program.T {functions, handlesSignals, main, objectTypes, profileI
       val {args, blocks, name, raises, returns, start} =
          Function.dest (insert main)
       val newStart = Label.newNoname ()
+      fun define x = Statement.Bind
+         {dst = (x, Type.cpointer ()),
+          pinned = false,
+          src = Operand.Static
+            {static=Static.T
+               {data=Static.Data.Object [Static.Data.Word (WordX.one WordSize.bool)],
+                header=WordXVector.fromList ({elementSize=WordSize.objptr ()}, []),
+                location=Static.MutStatic},
+             ty=Type.cpointer ()}}
```

Comment:
I think there's no downside to allowing access via a CPointer. I'd either leave this as is or delete it. No sense in optimizing it to an array.

Reply:
The downside of accessing as a `CPointer` to a raw boolean (as opposed to an `Objptr` to a `bool ref`) is that it would not be compatible with a future revision of static allocation that expects to be able to emit statics as a contiguous heap.

Reply:
I don't see why that would be incompatible. We can always have a mix of statically-allocated and heap data, and we should always prefer statically-allocated.

I just can't see any incompatibilities or downside. It's absolutely always safe and preferable to do.

Reply:
It would be incompatible if we wanted the collection of immutable and mutable statics to look enough like a GCed heap so that runtime functions like `foreachObjptrInRange` would work.  (For example, we could assert that each objptr in the "heaps" of immutable and mutable statics is an objptr into the "heaps" of statics; i.e., that there are no "down" pointers to the dynamic heap.)  This would require that the "heaps" of statics are exactly a contiguous sequence of *objects* (with proper headers, etc.).  That's what would be incompatible with a raw (non-object) boolean in this "heap" of statics and a `CPointer` directly to it.

Reply:
Okay, that's a fair point. It's certainly not too difficult to give it a proper header, and although there's some benefit to having CPointers be available if we're not doing anything to create extras you're definitely right that the invariant checking outweighs it.

Reply:
Though, it would still want to be kept as a CPointer for the sake of other GCs (or we remove tha pass). I'll still contend that there's no downside to allowing a CPointer into statics, (although there's relatively little benefit to keeping the pass around).

Reply:
I don't understand what you mean by "want to be kept as a CPointer for the sake of other GCs".  Just that by not treating it as an `Objptr`, then it is essentially invisible to GCs?

Reply:
Yes. The only time it matters if it someone happens to construct these and a GC could hypothetically not test if something's outside the heap (though on a contiguous heap that's a dead use case). Arguably one could save a miniscule amount of GC time by converting to CPointers anything that can never point to something that's collectable.